### PR TITLE
Create sql data warehouse with transparent data encryption

### PR DESCRIPTION
**Pull request by cmendible**
### Changelog

*  Create sql data warehouse with transparent data encryption

### CODE REVIEW

**Comment by cmendible** [comment]
Weird timeout in the Travis build. I successfully deployed the DTW in westeurope with the values shown in the the log by running: `az group deployment create --resource-group template --template-file azuredeploy.json --parameters azuredeploy.parameters.json`

output was:

`
{
  "id": "/subscriptions/[subscription id]/resourceGroups/template/providers/Microsoft.Resources/deployments/azuredeploy",
  "name": "azuredeploy",
  "properties": {
    "additionalProperties": {
      "duration": "PT4M39.052999S",
      "outputResources": [
        {
          "id": "/subscriptions/[subscription id]/resourceGroups/template/providers/Microsoft.Sql/servers/ci2876ac3ae80d72f7",
          "resourceGroup": "template"
        },
        {
          "id": "/subscriptions/[subscription id]/resourceGroups/template/providers/Microsoft.Sql/servers/ci2876ac3ae80d72f7/databases/ci37828791f593dc75",
          "resourceGroup": "template"
        },
        {
          "id": "/subscriptions/[subscription id]/resourceGroups/template/providers/Microsoft.Sql/servers/ci2876ac3ae80d72f7/databases/ci37828791f593dc75/transparentDataEncryption/current",
          "resourceGroup": "template"
        }
      ],
      "templateHash": "18153730212806317741"
    },
    "correlationId": "b966fb61-f6db-4ab9-8af2-06817acefb86",
    "debugSetting": null,
    "dependencies": [
      {
        "dependsOn": [
          {
            "id": "/subscriptions/[subscription id]/resourceGroups/template/providers/Microsoft.Sql/servers/ci2876ac3ae80d72f7",
            "resourceGroup": "template",
            "resourceName": "ci2876ac3ae80d72f7",
            "resourceType": "Microsoft.Sql/servers"
          }
        ],
        "id": "/subscriptions/[subscription id]/resourceGroups/template/providers/Microsoft.Sql/servers/ci2876ac3ae80d72f7/databases/ci37828791f593dc75",
        "resourceGroup": "template",
        "resourceName": "ci2876ac3ae80d72f7/ci37828791f593dc75",
        "resourceType": "Microsoft.Sql/servers/databases"
      },
      {
        "dependsOn": [
          {
            "id": "/subscriptions/[subscription id]/resourceGroups/template/providers/Microsoft.Sql/servers/ci2876ac3ae80d72f7/databases/ci37828791f593dc75",
            "resourceGroup": "template",
            "resourceName": "ci2876ac3ae80d72f7/ci37828791f593dc75",
            "resourceType": "Microsoft.Sql/servers/databases"
          }
        ],
        "id": "/subscriptions/[subscription id]/resourceGroups/template/providers/Microsoft.Sql/servers/ci2876ac3ae80d72f7/databases/ci37828791f593dc75/transparentDataEncryption/current",
        "resourceGroup": "template",
        "resourceName": "ci2876ac3ae80d72f7/ci37828791f593dc75/current",
        "resourceType": "Microsoft.Sql/servers/databases/transparentDataEncryption"
      }
    ],
    "mode": "Incremental",
    "outputs": null,
    "parameters": {
      "dataWarehouseName": {
        "type": "String",
        "value": "ci37828791f593dc75"
      },
      "serviceLevelObjective": {
        "type": "String",
        "value": "DW400"
      },
      "sqlAdministratorLogin": {
        "type": "String",
        "value": "cief1dd328ecf8dca7"
      },
      "sqlAdministratorPassword": {
        "type": "SecureString"
      },
      "sqlServerName": {
        "type": "String",
        "value": "ci2876ac3ae80d72f7"
      },
      "transparentDataEncryption": {
        "type": "String",
        "value": "Enabled"
      }
    },
    "parametersLink": null,
    "providers": [
      {
        "id": null,
        "namespace": "Microsoft.Sql",
        "registrationState": null,
        "resourceTypes": [
          {
            "aliases": null,
            "apiVersions": null,
            "locations": [
              "westeurope"
            ],
            "properties": null,
            "resourceType": "servers"
          },
          {
            "aliases": null,
            "apiVersions": null,
            "locations": [
              "westeurope"
            ],
            "properties": null,
            "resourceType": "servers/databases"
          },
          {
            "aliases": null,
            "apiVersions": null,
            "locations": [
              null
            ],
            "properties": null,
            "resourceType": "servers/databases/transparentDataEncryption"
          }
        ]
      }
    ],
    "provisioningState": "Succeeded",
    "template": null,
    "templateLink": null,
    "timestamp": "2018-05-18T12:12:39.631584+00:00"
  },
  "resourceGroup": "template"
}
`

Can we force the test again?

**Comment by cmendible** [comment]
Hi @bmoore-msft wondering if you can help me with this one.

Thanks!

File: 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "sqlServerName": {
+      "type": "string",
+      "metadata": {
```
Comment (bmoore-msft) [inline 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json]: generate a defaultValue, e.g.

"defaultValue": "[concat('sql', uniqueString(resourceGroup().id))]"

File: 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        "description": "Performance Level"
+      }
+    }
+  },
+  "variables": {},
+  "resources": [
+    {
+      "name": "[parameters('sqlServerName')]",
+      "type": "Microsoft.Sql/servers",
+      "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json]: update apiVersion

File: 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        "administratorLogin": "[parameters('sqlAdministratorLogin')]",
+        "administratorLoginPassword": "[parameters('sqlAdministratorPassword')]",
+        "version": "12.0"
+      },
+      "resources": [
+        {
+          "type": "databases",
+          "kind": "v12.0,user,datawarehouse",
+          "name": "[parameters('dataWarehouseName')]",
+          "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json]: update apiVersion

File: 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          },
+          "dependsOn": [
+            "[parameters('sqlServerName')]"
+          ],
+          "resources": [
+            {
+              "comments": "Transparent Data Encryption",
+              "name": "current",
+              "type": "transparentDataEncryption",
+              "apiVersion": "2014-04-01-preview",
```
Comment (bmoore-msft) [inline 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.json]: update apiVersion

File: 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        "sqlAdministratorLogin": {
+            "value": "GEN-UNIQUE"
+        },
+        "sqlAdministratorPassword": {
+            "value": "GEN-PASSWORD"
+        },
+        "dataWarehouseName": {
+            "value": "GEN-UNIQUE"
+        },
+        "transparentDataEncryption": {
```
Comment (bmoore-msft) [inline 201-sql-data-warehouse-transparent-encryption-create/azuredeploy.parameters.json]: remove the non-GEN values and rely on the defaulValues in the template

**Comment by cmendible** [comment]
@bmoore-msft thanks for the review. Hope c447d67 covers evrything...

**Pull request merged**